templates/song_matches.html

```diff
@@ -0,0 +1,77 @@
+{% extends "base.html" %}
+
+{% block title %}Matching Songs{% endblock %}
+
+{% block content %}
+
+    <h1>Here's what we got:</h1>
+
+
+    {% if mood %}
+        <h2>Matching Songs for {{ mood }} Mood</h2>
+    {% elif activity and city and weather_stats %}
+        <h2>Matching Songs for {{ activity }} in {{ city }}</h2>
+        <p>Weather: {{ weather_stats[0] }}°F, {{ weather_stats[1] }}</p>
+    {% elif original_song %}
+        <h2>Matching Similar Songs to {{ original_song }}</h2>
+    {% endif%}
+
+    <ul>
+        {% for key, song in songs.items() %}
+            <li>{{ song['song_name'] }} by {{ song['artist_name'] }} from the album {{ song['album_name'] }}
+            <br>
+            <a href="{{ song['song_link'] }}" target="_blank">Listen on Spotify</a>
+            <form class="save-song-form" data-song-id="{{ key }}" action="{{ url_for('song_matches') }}" method="POST">
+              <input type ="hidden" name="song_index" value="{{ key }}">
+              <input type="hidden" name="song_name" value="{{ song['song_name'] }}">
+              <input type="hidden" name="artist_name" value="{{ song['artist_name'] }}">
+              <input type="hidden" name="album_name" value="{{ song['album_name'] }}">
+              <input type="hidden" name="song_link" value="{{ song['song_link'] }}">
+              <button type="submit" class="btn btn-primary save-song-button">Save Song</button>
+            </form>
+            </li>
+        {% endfor %}
+    </ul>
+
+
+    {% if mood %}
+        <a href="{{ url_for('match_the_mood_info') }}">Change the parameters</a><br>
+    {% elif activity and city and weather_stats %}
+        <a href="{{ url_for('match_the_day_info') }}">Change the parameters</a><br>
+    {% elif original_song %}
+        <a href="{{ url_for('match_the_song_info') }}">Change the parameters</a><br>
+    {% endif%}
+    <a href="{{ url_for('discover') }}">Change the feature</a><br>
+    <a href="{{ url_for('saved_songs') }}">View Saved Songs</a><br>
+    <a href="{{ url_for('dashboard') }}">Back to dashboard</a>
+
+    <script>
```

Comment:
Looks like this JavaScript code block works. Would like to see this done in a separate JS file. Perhaps make a folder for that and then link that to here. Let me know if you need help with that.